addons/test_mail/tests/test_message_post.py

```diff
@@ -36,6 +41,19 @@ def test_message_post_return(self):
         messageId = call_kw(self.env['mail.channel'], 'message_post', [test_channel.id], {'body': 'test'})
         self.assertTrue(isinstance(messageId, int))
 
+    @users('demo')
+    def test_mail_add_signature(self):
+        user = self.env.user
+        self.test_track.user_id = user
+
+        msg_true = self.test_track.message_post(body="Test body", mail_auto_delete=False, add_sign=True, partner_ids=[self.partner_1.id, self.partner_2.id], email_layout_xmlid="mail.mail_notification_paynow")
+        found_mail = self.env['mail.mail'].sudo().search([('mail_message_id', '=', msg_true.id)])
+        self.assertTrue("Best regards" in str(found_mail.body_html))
```

Comment:
Note that it does not check that the signature is not present twice ... as a side effect it works but it could break again if we change a bit the underlying logic. You should check the signature present in the email and check that it is present only once (and run the test without the fix to be sure it fails, sometimes tests with html content are hard to write ^^ ).